### PR TITLE
feat(anta.tests): Adding the test case to verify BGP timers

### LINKED ISSUE #509: Add the test case to verify the BGP hold and keep alive timers

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
# Description

Adding the test case to verify BGP timers

Fixes #509 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/routing/bgp.py [hunk 1]
```diff
@@ -447,3 +448,71 @@ def test(self) -> None:
...
+
+        # Iterate over each bgp peer
+        for bgp_peer in self.inputs.bgp_peers:
+            peer_address = str(bgp_peer.peer)
+            vrf = bgp_peer.vrf
+            hold_time = bgp_peer.hold_time
+            keep_alive_time = bgp_peer.keep_alive_time
+
+            # Verify BGP peer
+            if not (bgp_output := get_value(self.instance_commands[0].json_output, f"vrfs..{vrf}..peerList", separator="..")):
```
Comment (carl-baillargeon) [inline anta/tests/routing/bgp.py:500]: Do you need to use `separator=".."` here?

File: anta/tests/routing/bgp.py [hunk 2]
```diff
@@ -447,3 +448,71 @@ def test(self) -> None:
...
+                failures[str(peer_address)] = {vrf: "Not configured"}
```
Comment (carl-baillargeon) [inline anta/tests/routing/bgp.py:501]: `peer_address` is already a `str` no?

File: anta/tests/routing/bgp.py [hunk 3]
```diff
@@ -447,3 +448,71 @@ def test(self) -> None:
...
+    Verifies if the BGP peers are configured with the correct hold and keep-alive timers in the specified VRF.
+    Expected results:
+        * success: The test will pass if the hold and keep-alive timers are correct for BGP peers in the specified VRF.
+        * failure: The test will fail if BGP peers are not found or hold and keep-alive timers are not correct in the specified VRF.
+    """
+
+    name = "VerifyBGPTimers"
+    description = "Verifies if the BGP peers are configured with the correct hold and keep alive timers in the specified VRF."
+    categories = ["routing", "bgp"]
+    commands = [AntaCommand(command="show bgp neighbors")]
```
Comment (carl-baillargeon) [inline anta/tests/routing/bgp.py:464]: `show bgp neighbors` will give you the neighbors of the default VRF only. You should use `show bgp neighbors vrf all` or use `AntaTemplate` and `show bgp neighbors vrf {vrf}`.

File: anta/tests/routing/bgp.py [hunk 4]
```diff
@@ -447,3 +448,71 @@ def test(self) -> None:
...
+            vrf = bgp_peer.vrf
+            hold_time = bgp_peer.hold_time
+            keep_alive_time = bgp_peer.keep_alive_time
+
+            # Verify BGP peer
+            if not (bgp_output := get_value(self.instance_commands[0].json_output, f"vrfs..{vrf}..peerList", separator="..")):
+                failures[str(peer_address)] = {vrf: "Not configured"}
+                continue
+
+            bgp_index = create_index(bgp_output, "peerAddress")
```
Comment (carl-baillargeon) [inline anta/tests/routing/bgp.py:504]: Can you check if the [get_item](https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/plugins/plugin_utils/utils/get_item.py) function in AVD does what you want? If so, let's copy the code in `anta.tools` so we can use it.

File: anta/tests/routing/bgp.py [hunk 5]
```diff
@@ -447,3 +448,71 @@ def test(self) -> None:
...
+            bgp_output = bgp_index.get(peer_address)
+            if not bgp_output:
+                failures[str(peer_address)] = {vrf: "Not configured"}
```
Comment (carl-baillargeon) [inline anta/tests/routing/bgp.py:507]: Why do we need a second statement here if there peer is not configured?

**Review by carl-baillargeon — CHANGES_REQUESTED** [review]

File: anta/tests/routing/bgp.py [hunk 6]
```diff
@@ -447,3 +448,67 @@ def test(self) -> None:
...
+        class BgpPeers(BaseModel):
+            """
+            This class defines the details of a BGP peer.
+            """
+
+            peer_address: IPv4Address
+            """IPv4 address of a BGP peer"""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+            hold_time: int
```
Comment (carl-baillargeon) [inline anta/tests/routing/bgp.py:483]:
```suggestion
            hold_time: int = Field(ge=3, le=7200)
```

File: anta/tests/routing/bgp.py [hunk 7]
```diff
@@ -447,3 +448,67 @@ def test(self) -> None:
...
+            """BGP hold time in seconds"""
+            keep_alive_time: int
```
Comment (carl-baillargeon) [inline anta/tests/routing/bgp.py:485]:
```suggestion
            keep_alive_time: int = Field(ge=0, le=3600)
```

**Review by carl-baillargeon — APPROVED** [review]
LGTM after restricting `int` inputs.

**Review by titom73 — APPROVED** [review]

**Pull request merged**